pkgs/top-level/static.nix

```diff
@@ -151,4 +155,19 @@ in {
     };
   };
 
+  aws-sdk-cpp = super.aws-sdk-cpp.override {
+    staticOnly = true;
+  };
+
+  brotli = super.brotli.override {
+    staticOnly = true;
+  };
+
+  curl = super.curl.override {
+    gssSupport = false;
+  };
+
+  nix = super.nix.overrideAttrs (_: {
+    NIX_LDFLAGS = "-lssl -lbrotlicommon -lssh2 -lz -lnghttp2 -lcrypto";
```

Comment:
I think the right way to do this would be patching the pkg-config files to include these libraries in the `Requires.private` section, as is done in:

```
$ nix-build -A brotli.dev && cat result-dev/lib/pkgconfig/libbrotlidec.pc
prefix=/nix/store/7i965pyrkqph3gwpfzswz35c6a1j1v5h-brotli-1.0.7
exec_prefix=/nix/store/7i965pyrkqph3gwpfzswz35c6a1j1v5h-brotli-1.0.7
libdir=/nix/store/iahrxydkznvf6z0q0hp0cp9nhrq1pkys-brotli-1.0.7-lib/lib
includedir=/nix/store/1qcykjlmxvvzgvvnyzsz7yii794m5rjv-brotli-1.0.7-dev/include

Name: libbrotlidec
URL: https://github.com/google/brotli
Description: Brotli decoder library
Version: 1.0.7
Libs: -L${libdir} -lbrotlidec
Requires.private: libbrotlicommon >= 1.0.2
Cflags: -I${includedir}
```
I haven't looked at this specific case too closely, but in general invoking `pkgconfig --static <args>` will also include the `Requires.private` sections for transitive link dependencies that are not needed in the dynamic variant.

Reply:
I think the issue was that all of these need to be propagated in static. static linking requires knowing where all of the needed libraries are.